flowey/flowey_core/src/github_context.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Core types and traits used to read GitHub context variables.
+
+use crate::node::{user_facing::GhContextVar, ClaimVar, NodeCtx, ReadVar, StepCtx};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+#[derive(Serialize, Deserialize)]
+pub struct Head {
+    #[serde(rename = "ref")]
+    pub head_ref: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct GhContextVarReaderEventPullRequest {
+    pub head: Head,
+}
+
+pub enum Root {}
```

Comment:
you still want to stuff these ZSTs into their own `mod state` inline module, just to avoid polluting this namespace with generic-sounding things like `Root` and `Event`